sympy/physics/continuum_mechanics/beam.py

```diff
@@ -946,8 +946,8 @@ def solve_for_reaction_loads(self, *reactions):
         rotation_jumps = tuple(self._rotation_hinge_symbols)
         deflection_jumps = tuple(self._sliding_hinge_symbols)
 
-        shear_curve = limit(self.shear_force(), x, l)
-        moment_curve = limit(self.bending_moment(), x, l)
+        shear_curve = limit(self.shear_force().rewrite('Heaviside'), x, l)
+        moment_curve = limit(self.bending_moment().rewrite('Heaviside'), x, l)
```

Comment:
I don't think that we want to rewrite as heaviside here. Rather limit should be fixed to work with singularity functions.